src/main/java/com/mlaf/hu/decisionagent/DecisionAgent.java

```diff
@@ -0,0 +1,102 @@
+package com.mlaf.hu.decisionagent;
+
+import com.mlaf.hu.decisionagent.behavior.ReceiveBehavior;
+import com.mlaf.hu.decisionagent.behavior.RegisterBehavior;
+import com.mlaf.hu.decisionagent.behavior.UpdateBehavior;
+import com.mlaf.hu.helpers.JadeServices;
+import com.mlaf.hu.models.*;
+import jade.core.AID;
+import jade.core.Agent;
+import jade.domain.DFService;
+import jade.util.Logger;
+
+import javax.xml.bind.JAXB;
+import java.io.StringReader;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public abstract class DecisionAgent extends Agent {
+    private static final String SERVICE_NAME = "DECISION-AGENT";
+    private static final int MAX_READINGS = 100;
+    public static java.util.logging.Logger decisionAgentLogger = Logger.getLogger("DecisionAgentLogger");
+    public HashMap<AID, InstructionSet> sensorAgents = new HashMap<>();
+
+    public DecisionAgent() {
+        super();
+    }
+
+    @Override
+    protected void setup() {
+        try {
+            JadeServices.registerAsService(SERVICE_NAME, "decision-agent", null, null, this);
+            addBehaviour(new RegisterBehavior(this));
+            addBehaviour(new ReceiveBehavior(this));
+            addBehaviour(new UpdateBehavior(this));
```

Comment:
Misschien kan dit beter een ticker zijn ofzo? Ik weet niet of we dit continu hoeven te updaten

Reply:
Welke van de behaviors? Of allemaal?

Reply:
Updatebehavior specefiek

Reply:
Hoeveel MS voor de ticker stel je voor?